### PR TITLE
Add appProtocol to Service Ports

### PR DESCRIPTION
**Pull request by shaardie**
This patch adds the appProtocol fields to the service ports. Some tools, like istio's service mesh are using those to specify how to handle the traffic.

Signed-off-by: Sven Haardiek <sven.haardiek@uni-muenster.de>

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@desaintmartin could you please take a look?

**Review by pierluigilenoci — CHANGES_REQUESTED** [review]

File: helm/oauth2-proxy/templates/service.yaml [hunk 1]
```diff
@@ -34,10 +34,12 @@ spec:
       nodePort: {{ .Values.service.nodePort }}
       {{- end }}
       protocol: TCP
+      appProtocol: http
```
[the remark this answers is not in the document]
Comment (shaardie) [reply helm/oauth2-proxy/templates/service.yaml:37]: Can this be something else in oauth2-proxy?

[the remark this answers is not in the document]
Comment (pierluigilenoci) [reply helm/oauth2-proxy/templates/service.yaml:37]: It depends, for example, if you inject a sidecar into the pod that might change.

[the remark this answers is not in the document]
Comment (shaardie) [reply helm/oauth2-proxy/templates/service.yaml:37]: funnily enough, this is exactly why I introduced this. We are using the istio cluster mesh and it identifies the kind of traffic either based on the prefixed name (e.g. http-something) or on `appProtocol`. Istio works perfectly with setting this to http and itself is hijacking the connection and introducing mtls.

That said, I am perfectly fine with making it configurable (and set the default to http(?)), if this means that we can get this upstream and have not to patch this into the helm chart ourself.

[the remark this answers is not in the document]
Comment (pierluigilenoci) [reply helm/oauth2-proxy/templates/service.yaml:37]: Please set the default as `http`

**Review by pierluigilenoci — APPROVED** [review]

**Comment by pierluigilenoci** [comment]
@shaardie I just realized now that the revision of the [README.md](https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/README.md) is missing. 
Please add the new parameter in the documentation.

**Comment by shaardie** [comment]
@pierluigilenoci I missed that. But added it now.

**Review by pierluigilenoci — DISMISSED** [review]

**Comment by shaardie** [comment]
I updated the Chart and resolved the merge conflicts.

**Review by pierluigilenoci — APPROVED** [review]

**Pull request merged**